### ISSUE TITLE
Stream source ISO files directly to Kodi, without transcoding

### ISSUE THREAD

**Issue by strugee** [jellyfin/jellyfin-kodi#430]
**Describe the bug**

I have a bunch of ISO files of DVDs that I've ripped. Jellyfin is currently basically incapable of playing these (my hardware is *way* too old for realtime transcoding and Jellyfin's ISO support isn't great anyway), but Kodi is - I verified this by adding the underlying source that Jellyfin is using (my personal Nextcloud) to Kodi directly. However, the Jellyfin addon for Kodi doesn't know this, and AFAICT tries to transcode the source file before streaming it to Kodi. This isn't great because a) I can't watch the movie at all, but b) even if I could I wouldn't get the menus and stuff because it would be transcoded to a video stream only.

**To Reproduce**

1. Have an ISO file from a DVD in your Jellyfin library
2. Go to Kodi home > Movies and pick the movie backed by an ISO file

**Expected behavior**

Kodi is handed an ISO to play and I see a DVD menu.

**Actual behavior**

I see the loading indicator for a few moments and then I'm booted back to the Movies menu. The media does not play.

**Logs**

https://paste.osmc.tv/uconopezan (I just tried to play Dead Poets Society to reproduce this in the logs, so if you want to do a find, search for `DEAD_POETS_SOCIETY.iso` since that's the backing filename)

**Screenshots**

N/A

**System (please complete the following information):**
 - OS: OSMC 2020.10-1 (based on Debian)
 - Jellyfin Version: 10.6.4
 - Kodi Version: 18.8
 - Addon Version: 0.6.3+py2 (can't figure out how to force a Kodi repo refresh which is why I don't have the release from earlier today)
 - Playback Mode: Add-On

**Additional context**

N/A

**Comment by mcarlton00** [comment]
To my knowledge, the only way this is remotely feasible will be to run the addon in Native mode.  Either the server has to remux the contents of the disk and send them to us as a stream, or you have to transfer the entire file over the network before it can be read.  ISO files are kinda an all or nothing deal when you're trying to read data from them.

**Comment by strugee** [comment]
Yeah, that definitely makes sense. But anecdotally, performance with Kodi's native ISO reading over WebDAV is extremely tolerable (at least on the 3 or so ISOs I tried). It took a few seconds for the DVD menu to load and that was it. Either Kodi is doing something really clever and doing a ton of range requests to get only the parts of the ISO it needs, or (far more likely) the Jellyfin server's network and my current network are simply fast enough that this is a non-issue. Note that both networks are wired, so that supports the theory that it's just fast enough to not be a problem.

To clarify, are you saying I should try switching the addon to native mode and see if it works then?

**Comment by mcarlton00** [comment]
That's the only way I can personally see this working how you want.  It hasn't really been designed in a way to facilitate raw iso files.  Even when using webdav you still have to retrieve the whole file, or at a minimum do a lot of processing to know what pieces you can get by with at a time.

**Comment by strugee** [comment]
> I see the loading indicator for a few moments and then I'm booted back to the Movies menu. The media does not play.

I still see this behavior in direct mode.

In the future I could see Jellyfin potentially being able to read the ISO file itself and stream the contents in a more intelligent way, but since that functionality doesn't exist yet, downloading the whole ISO seems like a nice compromise (at least, on my admittedly very fast network).

**Comment by Artiume** [comment]
Can you try extracting the ISO and reading that. i think Kodi is mounting the ISO and reading the VOB and IFO files directly

**Comment by mmortal03** [comment]
@strugee , are you still seeing this? I'm currently testing on Kodi 19.1 an ISO image of a DVD with complex menus shared over SMB. Menus seem to be working properly. I'm only noticing some possible bugginess in the Kodi playback controls not working as expected in certain specific contexts, but that could be a separate bug.
